xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -1070,27 +1099,6 @@ private void close(Exception error) {
 
```

Comment:
You should also call `cleanUpResources()` here?

Reply:
Wait... Are you commenting on the code including the last commit? It shows as "outdated".

We don't call `XdsClientImpl#cleanUpResources()` inside `AdsStream#close()`. Now `cleanUpResources()` is `XdsClientImpl`'s state cleanup logic. In `XdsClientImpl#shutdown()`, it closes the RPC stream and calls `cleanUpResources()` to clean up cache and cancel response timers. In `AdsStream#handleStreamClosed()`, it destroys the current RPC stream and calls `XdsClientImpl#cleanUpResources()` as well.